lib/SIL/Verifier/MemoryLifetimeVerifier.cpp

```diff
@@ -556,6 +556,12 @@ void MemoryLifetimeVerifier::checkBlock(SILBasicBlock *block, Bits &bits) {
             break;
           case StoreOwnershipQualifier::Trivial:
             locations.setBits(bits, SI->getDest());
+            // Trivial values may be stored in locations with a nontrivial type.
+            // When this happens, deinitialization is optional.
+            //
+            // TODO: Fix SILGen so that nontrivial memory locations can never
+            // have a trivial value (borrowed memory is just as efficient).
+            locations.clearNonTrivialLocation(SI->getDest());
```

Comment:
Why is this needed? This should be handled by https://github.com/apple/swift/blob/72ff115284a2b21193296648b9b663118594dd41/lib/SIL/Verifier/MemoryLifetimeVerifier.cpp#L144